### ISSUE TITLE
MD034: Email autolink with incomplete domain in angle brackets incorrectly reported as a violation

### ISSUE THREAD

**Issue by MichaelTurekCGI** [DavidAnson/markdownlint#1140]
Similar to: https://github.com/igorshubovych/markdownlint-cli/issues/339

When running the `markdownlint-cli2` Pre-commit hook with the `--fix` option, a pair of angled brackets is added around text that is detected as a URL infinitely.

## Minimal Markdown Example:

```markdown
# Foo

Bar compose@.service

```

## Expected behavior:

Run `N`:
```markdown
# Foo

Bar compose@.service

```

## Actual behavior:

Run 1:
```markdown
# Foo

Bar <compose@.service>

```

Run 2:
```markdown
# Foo

Bar <<compose@.service>>

```

Run 3:
```markdown
# Foo

Bar <<<compose@.service>>>

```

et c.

**Comment by DavidAnson** [comment]
```markdown
OK: <user@example.com>

Bad: <user@.com>
```
